crates/uv-client/src/registry_client.rs

```diff
@@ -1136,9 +1182,82 @@ impl SimpleMetadata {
                 }
             }
         }
+
         Self(
-            map.into_iter()
-                .map(|(version, files)| SimpleMetadatum { version, files })
+            version_map
+                .into_iter()
+                .map(|(version, files)| SimpleMetadatum {
+                    version,
+                    files,
+                    metadata: None,
+                })
+                .collect(),
+        )
+    }
+
+    fn from_pyx_files(
+        files: Vec<uv_pypi_types::PyxFile>,
+        mut core_metadata: FxHashMap<Version, uv_pypi_types::CoreMetadatum>,
+        package_name: &PackageName,
+        base: &Url,
+    ) -> Self {
+        let mut version_map: BTreeMap<Version, VersionFiles> = BTreeMap::default();
+
+        // Convert to a reference-counted string.
+        let base = SmallString::from(base.as_str());
+
+        // Group the distributions by version and kind
+        for file in files {
+            let file = match File::try_from_pyx(file, &base) {
+                Ok(file) => file,
+                Err(err) => {
+                    // Ignore files with unparsable version specifiers.
+                    warn!("Skipping file for {package_name}: {err}");
+                    continue;
+                }
+            };
+            let Some(filename) = DistFilename::try_from_filename(&file.filename, package_name)
+            else {
+                warn!("Skipping file for {package_name}: {}", file.filename);
```

Comment:
Assuming this matches an existing pattern but it'd be nice to say why this is skipped in the future.